gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,309 @@
+    String clientBuilderClassName = configuration.get("nessie.client-builder-impl");
+    if (clientBuilderClassName == null) {
+      // Use default HttpClientBuilder
+      return HttpClientBuilder.builder().fromConfig(configuration::get).build(NessieApiV1.class);
+    }
+    // Use custom client builder
+    NessieClientBuilder builder;
+    try {
+      builder =
+          (NessieClientBuilder)
+              Class.forName(clientBuilderClassName).getDeclaredConstructor().newInstance();
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException(
+          String.format("No custom client builder class found for '%s' ", clientBuilderClassName));
+    } catch (InvocationTargetException
+        | InstantiationException
+        | IllegalAccessException
+        | NoSuchMethodException e) {
+      throw new IllegalArgumentException(
+          String.format("Could not initialize '%s': ", clientBuilderClassName), e);
+    }
+    return (NessieApiV1) builder.fromConfig(configuration::get).build(NessieApiV1.class);
+  }
+
+  /**
+   * Compute the bloom filter per content id by walking all the live references in a distributed way
+   * using spark.
+   *
+   * @param references list of live references to walk.
+   * @param cutOffTimestampPerRef map of cutoff timestamp per reference.
+   * @param defaultCutOffTimestamp default cutoff timestamp to be used.
+   * @return map of {@link ContentBloomFilter} per content-id.
+   */
+  public Map<String, ContentBloomFilter> getLiveContentsBloomFilters(
+      List<Reference> references,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp) {
+    List<Map<String, ContentBloomFilter>> bloomFilterMaps =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, getTaskCount(references))
+            .map(
+                reference ->
+                    computeLiveContents(
+                        config, cutOffTimestampPerRef, defaultCutOffTimestamp, reference))
+            .collect();
+    return mergeLiveContentResults(bloomFilterMaps);
+  }
+
+  /**
+   * Gets the live and expired contents per content id by walking all the live and dead references
+   * in a distributed way using spark and checking the contents against the live bloom filter
+   * results.
+   *
+   * @param liveContentsBloomFilterMap live contents bloom filter per content id.
+   * @param references list of all the references to walk (live and dead)
+   * @return {@link IdentifiedResult} object.
+   */
+  public IdentifiedResult getIdentifiedResults(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
+
+    List<IdentifiedResult> results =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, getTaskCount(references))
+            .map(reference -> computeExpiredContents(config, liveContentsBloomFilterMap, reference))
```

Comment:
No need to pass the instance field `config` to the instance method `computeExpirecContents`